### ISSUE TITLE
Can't reopen a World

### ISSUE THREAD

**Issue by Switch-Back18** [quek04/The-Undergarden#60]
I have install you mod and after i wan't to create a new world and i can't reopen it

[05janv.2021 12:53:36.080] [Render thread/WARN] [net.minecraft.client.Minecraft/]: Failed to load datapacks, can't proceed with server load
java.lang.IllegalStateException: Failed to load world

When i uninstall it the bug is fix !

Could you fix it ?

**Comment by remplerus** [comment]
what version do you use?

**Comment by remplerus** [comment]
But you are right, i have the issue too with version 0.4.0 and 0.4.1, with 0.3.8 i can join my world again.
i get the following "crash log" when i join the world (thanks to Blame i can see what is causing the crash):
https://gist.github.com/remplerus/f6423a1c39a287e429b68a49a4e6fd76

**Comment by ghost** [comment]
Does removing undergarden work to resolve the issue too

**Comment by Switch-Back18** [comment]
Yes

**Comment by TheRedNeckGamer** [comment]
The issue is that you're using other worldgen mods and frankly, undergarden isn't implemented correctly. Catacombs are not registered to the structures map.

**Comment by Switch-Back18** [comment]
Yes, i use Garden of Glass from Botania, will you fix that later or not ?

**Comment by Digimasterhir0** [comment]
![271658858_Sinttulo](https://user-images.githubusercontent.com/47353693/104827477-144ae980-586f-11eb-9294-bbac235086d9.png)
I couldn't reenter the world with this error. After I've removed the Undergarden mod, everything became fine.

**Comment by remplerus** [comment]
downgrade to 0.3.* and it should work

**Comment by TheRedNeckGamer** [comment]
Not even a little bit fam. Now I can't even make it past the forge injection screen. It's fine. I'll play the other 390+ mods that were written correctly. 
![image](https://user-images.githubusercontent.com/72539092/104836327-cdf78980-5872-11eb-82e5-d5e6e28fc9cd.png)

**Comment by JulianPlusPlus** [comment]
take your trash talk somewhere else please :)

**Comment by Fooshcookie** [comment]
Don't worry about them, Julian, honestly. They're just upset they don't have their new micro-LED fan yet. A lack of flamboyant rainbow cooling makes gamer angry. Not like they need more, it's just another way they can go flex their lack of dignity.

**Comment by TheRedNeckGamer** [comment]
You're actually both wrong and I've already torn his source code apart and fixed the errors(both of them) on my end. I was trying to be helpful. I even pointed out that they were indeed two separate issues. I'm a shadow-developer, as in, I -fix- code and cross-compatibility for mods/packs. and fyi... I do all of this from a shitty 2015 dell laptop... <I don't even play so much as debug/build packs for mates>

If you fix the problem as opposed to avoiding it, and sending people to attempt an older version with even less compatibility, you'll get further in life.

**Comment by Fooshcookie** [comment]
Look, I'll be clear. If you're going to help, help. If you're not going to help, then don't. Don't say 'well, ackshulley'. That gets us no-where. Speaking how fast you were willing to just run off with "I'll play the other 390+ mods that were written correctly", I highly doubt you actually did anything with the mod.

If you do in-fact have the expertise to be fixing the issue, suggest a fix to the big man or provide examples. I can see why you stay 'in the shadows', you'd be -adored- by developers. The person you need to contact for that is quek, the only person who can get a fix out is quek; if there is something wrong, you need quek. The person recommending the old code, isn't quek, nor anyone who works on the mod (edit: I believe). Nor am I, I simply follow the mod closely.

Speaking quek also handles other mods, it'd be best to do the whole community a favor, and as cliche as it is: walk the walk, instead of talking the talk. If there is something drastically wrong, you might wanna get his attention instead of making yourself look like a tool. I've had a truly -awful- week, and I don't want to sit here having to listen to this. 

Sure, I could just walk away, but I feel it's important the issue gets resolved so everyone can enjoy the mod. I apologize personally for the hostility, but you're not helping the thread get anywhere, as I don't even think Quek has seen this one, yet, let alone wants to read through you talking about your '390+ mods that were actually written correctly'."

TLDR: If you want to chit-chat or talk about your 506 confirmed kills as a Navy Seal, fine, do that in the discord; however if you have a fix, take it to the quek or try to fork a fix, or something. The others here can't do anything, including myself.

Edit 2: Honestly, ignore me, I could be talking out of my ass. My brain clearly isn't firing on all cylinders. Going to sleep.

**Comment by TheRedNeckGamer** [comment]
I was operating under the assumption that there was a development team to which the "remplerus" fellow belonged, the way he tried to pass the downgrade/"already addressed in another place" on me. I apologize if that's not the case. That said, I did indeed already fix the issue on my end, and am currently learning how to do so on github. 

TL;DR :
It's a 5 second fix and I was literally under the impression that "remplerus" had the ability to fix it but was too lazy. 

No offense was ever intended to quek04, nor implied.

**Comment by TheRedNeckGamer** [comment]
I was operating under the assumption that there was a development team to which the "remplerus" fellow belonged, the way he tried to pass the downgrade/"already addressed in another place" on me. I apologize if that's not the case. That said, I did indeed already fix the issue on my end, and am currently learning how to do so on github.

TL;DR :
It's a 5 second fix and I was literally under the impression that "remplerus" had the ability to fix it but was too lazy.

No offense was ever intended to quek04, nor implied.

EDIT: @quek04 I understand this is an alpha release (0.x.x), so yes, I absolutely called it an improperly coded mod. Clearly you are also aware of such or you would have called it a full or beta release (1.x.x) If you're offended in any way, I do apologize. If you're unable to fix the issue (I can't figure out github although I really should teach myself) =>UGmod/src/main/java/quek/undergarden/registry/UGStructures.java
A simple method edit is all that's necessary. Basically tell it to add to flat map as well as the default structure map.

**Comment by remplerus** [comment]
well just fyi im no mod dev. i cant fix anything. i just use debug for my modpack and exclude any updates. i just stick on to curseforge releases and no dev things. anyway, if you fixed it, it would be nice to make a PR to fix anything.

**Comment by quek04** [comment]
If you fixed the issue @TheRedNeckGamer , why don't you make a PR?

**Comment by TheRedNeckGamer** [comment]
As previously stated I don't know anything about github or how it operates. I literally told you how and where to fix it. I will be learning the layout of github and how to do these things but for now, I did the best I can. I'm new to Git. I've used dropbox for my stuff in the past and am just now returning to modding/pack building/coding. If you'd like to show me some of the ropes I'd be grateful tbh.

**Comment by ghost** [comment]
whaaaaaaaaaaaaaat I will friggin do it just tell me the code to change lmao and stop arguing

**Comment by MartyGentillon** [comment]
Basically, Git is version control software.  It exists so that when you mess something up, you can undo the change, even if you discover the mistake months after the fact.  It also helps with collaborative development when, for instance, two people change the same file at the same time.  Understanding Git is key to modern professional software development and also very useful for private projects.

In order to do these things, git uses a directed graph data structure to store the history of your project.  It then provides tools to manipulate that data structure: add new commits (nodes), copy changes around, and synchronize the data structure across multiple computers.  Any time you add a new version of your code to this data structure, you can return to it, forever.  Moreover, this allows you to track changes to individual files so that, if two people change the same file, you can detect and deal with the issue.

I would recommend the following tutorials as being pretty good for getting to grasps with the software:

Procedural tutorial: https://www.afternerd.com/blog/learn-git/
An interactive tutorial: https://learngitbranching.js.org/
The underlying structure: https://blog.jayway.com/2013/03/03/git-is-a-purely-functional-data-structure/
Undoing things with git: https://github.blog/2015-06-08-how-to-undo-almost-anything-with-git/
